### ISSUE TITLE
'+' PropType is defined but prop is never used

### ISSUE THREAD

**Issue by NullDivision** [yannickcr/eslint-plugin-react#1305]
I don't think immutable object props are supported.

Eslint: 3.19.0
Eslint-plugin-react: 7.0.1

**Comment by ljharb** [comment]
Can you provide sample code? What happens with eslint-plugin-react v7.1.0?

**Comment by NullDivision** [comment]
No change in v7.1.0.

```javascript
// @flow

import React from 'react';

type Props = {| +a: number, +b: string |};

const MyComp = ({ a, b }: Props) => <div>{a} <span>{b}</span></div>;

export default MyComp;
```

**Comment by ljharb** [comment]
What happens if you remove the "exact" pipes? Same error?

**Comment by alexzherdev** [comment]
This works now with the shared detection, but I think it was originally fixed in #1390.
